presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -1066,6 +1067,61 @@ protected Object visitLikePredicate(LikePredicate node, Object context)
                     optimizedEscape);
         }
 
+        private boolean isLikePattern(Slice pattern, Slice escape)
+        {
+            String stringPattern = pattern.toStringUtf8();
+            if (escape == null) {
+                return stringPattern.contains("%") || stringPattern.contains("_");
+            }
+
+            String stringEscape = escape.toStringUtf8();
+            if (stringEscape.length() != 1) {
+                throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Escape must be empty or a single character");
+            }
+            char escapeChar = stringEscape.charAt(0);
+            boolean escaped = false;
+            for (char currentChar : stringPattern.toCharArray()) {
```

Comment:
Actually, I think this works fine on characters. We never do anything that would split the surrogate pair.